Framework/DataHandling/src/LoadMuonNexusV2.cpp

```diff
@@ -145,20 +150,21 @@ void LoadMuonNexusV2::execLoader() {
   NXRoot root(m_filename);
   NXEntry entry = root.openEntry(NeXusEntry::RAWDATA);
   isEntryMultiPeriod(entry);
-  if (m_isFileMultiPeriod) {
-    throw std::invalid_argument(
-        "Multiperiod nexus files not yet supported by LoadMuonNexusV2");
-  }
-
   // Execute child algorithm LoadISISNexus2
   auto outWS = runLoadISISNexus();
 
   // Check if single or multi period file and create appropriate loading
   // strategy
   if (m_multiPeriodsLoaded) {
```

Comment:
This entire if block should be moved into a private method now such as `selectLoadStrategy`
You need to check your dynamic casts:  if they're "guaranteed" throw an `assert(workspaceGroup);` in so that future developers don't mess them up
If there is a chance they could fail for normal users then you need to check and throw